src/storage/heap_file.c

```diff
@@ -22098,6 +22100,7 @@ heap_update_bigone (THREAD_ENTRY * thread_p, HEAP_OPERATION_CONTEXT * context, b
   /* Fall through to exit. */
 
 exit:
+  log_append_empty_record (thread_p, LOG_END_ATOMIC_REPL, NULL);
```

Comment:
End atomic replication log record should always follow a start atomic replication log record. The code now allows one case that logs the end atomic replication, but does not log the start atomic replication, see the first `goto exit`.

Fix by:
- Move the start atomic replication log record before the first `goto exit`.
- Add a boolean to track if start atomic replication log was appended; append the end atomic replication log record only if the boolean is true.

Reply:
added a boolean flag to be inline with the other implementations